### PR TITLE
lncli: fix reader of unlock --stdin to read till EOF

### PR DESCRIPTION
**Pull request by DarthBenro008**
- fixes: #5584 

#### Pull Request Checklist

- [x] All changes are Go version 1.15 compliant
- [x] Your PR passes all CI checks. If a check cannot be passed for a justifiable reason, that reason must be stated in the commit message and PR description.
- [ ] If this is your first time contributing, we recommend you read the [Code Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md)
   - [ ] The code being submitted is commented according to [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation)
   - [ ] Commits have a logical structure according to [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure)
- [ ] For new code: Code is accompanied by tests which exercise both the positive and negative (error paths) conditions (if applicable)
- [ ] For bug fixes: If possible, code is accompanied by new tests which trigger the bug being fixed to prevent regressions
- [x] Any new logging statements use an appropriate subsystem and logging level
- [ ] For code and documentation: lines are wrapped at 80 characters (the tab character should be counted as 8 characters, not 4, as some IDEs do per default)
- [ ] A description of your changes [should be added to running the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes) for the milestone your change will land in.

### CODE REVIEW

File: cmd/lncli/cmd_walletunlocker.go [hunk 1]
```diff
@@ -495,7 +495,7 @@ func unlock(ctx *cli.Context) error {
 	// echoed in the console.
 	case ctx.IsSet("stdin"):
 		reader := bufio.NewReader(os.Stdin)
-		pw, err = reader.ReadBytes('\n')
+		pw, err = ioutil.ReadAll(reader)
```
Comment (carlaKC) [inline cmd/lncli/cmd_walletunlocker.go:498]: Below we trim newlines, seems like that could be an issue if a password ends with a newline?

**Review by carlaKC — COMMENTED** [review]
Just need to make sure that we're not going to trim newlines from passwords that intend to have them.

**Comment by DarthBenro008** [comment]
Nice catch!
I have removed the trim of newlines and ensured it works with a simple `fmt.Println` while debugging
PS: the text file i tested has two newlines

image

**Review by carlaKC — COMMENTED** [review]
This needs a commit in the 0.14 release notes as well! 

Just wondering what the impact of this change will be on the regular use case where a password is terminated by a newline?

eg, I type in my supersecure `p4ssw0rd123\n` (terminated with a newline by accident) and now we no longer trim that newline? Bit of a tradeoff here with the case where we do actually want a newline. Perhaps worth adding a warning if `pw != bytes.Trim(pw, "\n")`?

**Comment by carlaKC** [comment]
Closing due to inactivity.

**Comment by mohamedawnallah** [comment]
Replaced by #8657

**Pull request closed without merging**